File: pythoncollections/listprogram/sumofpairs.py
lst=[1,2,3,4,5,6,7,8]
lst.sort()
element=int(input("Enter the sum tht u want "))
low=0
upp=len(lst)-1
pair_lst=list()
while(low<upp):
    tot=lst[low]+lst[upp]
    if(element==tot):
        pair_lst.append((lst[low],lst[upp]))
        upp-=1
    elif(element<tot):
        upp-=1
    elif(element>tot):
        low+=1
print(pair_lst)

File: JAVASCRIPT/workshop_array/sumofpairs.js
arr=[1,2,3,4,5,6,7,8]
var sumlist=[]
var num=5
var low=0
var upp=arr.length-1
while(low<upp){
    tot=arr[low]+arr[upp]
    if(tot==num){
        sumlist.push(arr[low],arr[upp])
        upp-=1
    }
    else if(num<tot){
        upp-=1
    }
    else if(num>tot){
        low+=1
    }
}
console.log(sumlist);

